ctapipe/irf/irf_classes.py

```diff
@@ -0,0 +1,389 @@
+        default_value=None, directory_ok=False, help="Electron input filename and path"
+    ).tag(config=True)
+    electron_sim_spectrum = traits.Unicode(
+        default_value="IRFDOC_ELECTRON_SPECTRUM",
+        help="Name of the pyrif spectra used for the simulated electron spectrum",
+    ).tag(config=True)
+
+    chunk_size = Integer(
+        default_value=100000,
+        allow_none=True,
+        help="How many subarray events to load at once for making predictions.",
+    ).tag(config=True)
+
+    output_path = traits.Path(
+        default_value="./IRF.fits.gz",
+        allow_none=False,
+        directory_ok=False,
+        help="Output file",
+    ).tag(config=True)
+
+    overwrite = Bool(
+        False,
+        help="Overwrite the output file if it exists",
+    ).tag(config=True)
+
+    obs_time = Float(default_value=50.0, help="Observation time").tag(config=True)
+    obs_time_unit = Unicode(
+        default_value="hour",
+        help="Unit used to specify observation time as an astropy unit string.",
+    ).tag(config=True)
+
+    alpha = Float(
+        default_value=0.2, help="Ratio between size of on and off regions"
+    ).tag(config=True)
+    ON_radius = Float(default_value=1.0, help="Radius of ON region in degrees").tag(
+        config=True
+    )
+
+    max_bg_radius = Float(
+        default_value=5.0, help="Radius used to calculate background rate in degrees"
+    ).tag(config=True)
+
+    max_gh_cut_efficiency = Float(
+        default_value=0.8, help="Maximum gamma purity requested"
+    ).tag(config=True)
+    gh_cut_efficiency_step = Float(
+        default_value=0.1,
+        help="Stepsize used for scanning after optimal gammaness cut",
+    ).tag(config=True)
+    initial_gh_cut_efficency = Float(
+        default_value=0.4, help="Start value of gamma purity before optimisation"
+    ).tag(config=True)
+
+
+class EventPreProcessor(QualityQuery):
+    energy_reconstructor = Unicode(
+        default_value="RandomForestRegressor",
+        help="Prefix of the reco `_energy` column",
+    ).tag(config=True)
+    geometry_reconstructor = Unicode(
+        default_value="HillasReconstructor",
+        help="Prefix of the `_alt` and `_az` reco geometry columns",
+    ).tag(config=True)
+    gammaness_classifier = Unicode(
+        default_value="RandomForestClassifier",
+        help="Prefix of the classifier `_prediction` column",
+    ).tag(config=True)
+
+    preselect_criteria = List(
+        default_value=[
+            ("multiplicity 4", "subarray.multiplicity(tels_with_trigger) >= 4"),
+            ("valid classifier", "RandomForestClassifier_is_valid"),
+            ("valid geom reco", "HillasReconstructor_is_valid"),
+            ("valid energy reco", "RandomForestRegressor_is_valid"),
+        ],
+        help=QualityQuery.quality_criteria.help,
+    ).tag(config=True)
+
+    rename_columns = List(
+        help="List containing translation pairs new and old column names"
```

Comment:
So it's a mapping? then why not use a `Dict(value_trait=Unicode, key_trait=Unicode)`?

Reply:
This way it stays consistent with how you specify quality criteria, plus I don't have to rewrite the code :)